src/Joint.cc

```diff
@@ -149,7 +155,8 @@ Errors Joint::Load(ElementPtr _sdf)
         "The child element is missing."});
   }
 
-  if (this->dataPtr->childName == "world")
+  if (!_config.PermitWorldAsJointChild() &&
```

Comment:
I think of `ParserConfig` as a place where we put general parsing settings and that it's meant for end users. Here, I feel like it is being used to pass information from one DOM class to another. I don't think we want end users to call `PermitWorldAsJointChild`, so putting it in `ParserConfig` doesn't seem right. Would it be possible to just check if the `_sdf->parent()` is a world element instead?

Reply:
from a separate discussion, we decided it would be simpler to keep the same rule disallowing `world` in all instances of `//joint/child`, so I'll remove the configuration option

Reply:
configuration option removed in https://github.com/gazebosim/sdformat/pull/1117/commits/6ac126ef9cb475d81104f2360b23dafbc086d18e